rpc/RPCSession.java

```diff
@@ -97,4 +103,12 @@ public String database() {
                 return SessionProto.Session.Type.UNRECOGNIZED;
         }
     }
+
+    private class PulseTask extends TimerTask {
+        @Override
+        public void run() {
+            if (!isOpen()) return;
+            blockingGrpcStub.sessionPulse(SessionProto.Session.Pulse.Req.newBuilder().setSessionId(sessionId).build());
```

Comment:
This won't work by itself - we need to use the `blockingGrpcStub` to actually transmit the request.

The response from the server will then include a Boolean flag named `isAlive` that indicates if the session is still alive on the server. If it is not, we should close it in the client.

Reply:
Never mind I read it wrong